internal/transport/http_util.go

```diff
@@ -405,16 +406,28 @@ func newFramer(conn net.Conn, writeBufferSize, readBufferSize int, sharedWriteBu
 		pool = getWriteBufferPool(writeBufferSize)
 	}
 	w := newBufWriter(conn, writeBufferSize, pool)
-	f := &framer{
-		writer: w,
-		fr:     http2.NewFramer(w, r),
-	}
-	f.fr.SetMaxReadFrameSize(http2MaxFrameLen)
-	// Opt-in to Frame reuse API on framer to reduce garbage.
-	// Frames aren't safe to read from after a subsequent call to ReadFrame.
-	f.fr.SetReuseFrames()
-	f.fr.MaxHeaderListSize = maxHeaderListSize
-	f.fr.ReadMetaHeaders = hpack.NewDecoder(http2InitHeaderTableSize, nil)
+	f := &framer{writer: w}
+
+	if envconfig.GRPCHTTP2Framer == "EXPERIMENTAL" {
+		tfr := newGRPCFramer(w, r)
+		tfr.stubFramer.SetMaxReadFrameSize(http2MaxFrameLen)
```

Comment:
Could we implement `SetMaxReadFrameSize` on the GRPCFramer and so that this looks similar to the else case below. I believe, if we can achieve that, we could probably use generics to template the stuff here